network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -340,6 +411,16 @@ func (c *ControlMsgValidationInspector) processInspectMsgReq(req *InspectMsgRequ
 	}()
 
 	count := c.getCtrlMsgCount(req.validationConfig.ControlMsg, req.ctrlMsg)
+
+	// iWant validation uses new sample size validation. This will be updated for all other control message types.
+	switch req.validationConfig.ControlMsg {
+	case p2pmsg.CtrlMsgIWant:
+		if err := c.inspectIWant(req.ctrlMsg.GetIwant()); err != nil {
+			c.logAndDistributeAsyncInspectErr(req, count, err)
+		}
+		return nil
```

Comment:
Please note that the `inspectIWant` method might return errors other than `DuplicateFoundErr` and `IWantCacheMissThresholdErr`. It would be prudent to examine the specific error here and send notifications only for the ones we anticipate. Furthermore, in the context of calling `c.logAndDistributeAsyncInspectErr(req, count, err)`, the `count` in this context should represent the number of infractions, such as the number of duplicate message-ids or the number of cache misses, rather than a generalized count. This will ensure that the implementation aligns more closely with our intended functionality and error handling.

Reply:
Count handled in follow-up PR https://github.com/onflow/flow-go/pull/4642 

https://github.com/onflow/flow-go/pull/4574/commits/7721c00f37529994876508a43f1f81eff8c28956